Automattic/wp-calypso
# Checkout: display jetpack product conflict notices using new feature checks

#### Changes proposed in this Pull Request

* This PR replaces various checks within `sites/products/conflicts.ts` with siteHasFeature checks. The checks are related to showing the warning to users if they are purchasing the product to which they already have access. ie. The user might have bought a Jetpack scan and they try to buy a Jetpack Security plan. 
* This PR also removes unused functions after replacing the checks with the siteHasFeatures function. It also removes associated unit tests. 

#### Testing instructions



* Get a new Jetpack site from https://jurassic.ninja/ and register with Wordpress.com 
* Purchase jetpack scan
* Go to Calypso by visiting the link attached. 
* Select the newly created jetpack site from the sidebar
* Go to Upgrades > Plans and try to purchase Security Plan. 
* You see the following warning at the top of the shopping cart.
![](https://cloudup.com/imF7XRoDKwI+)

Repeat same tests for Backup and Antispam features. 

These tests are to show a warning message if the feature is already purchased individually when purchasing a plan. 

* Use the previously created site & remove all subscriptions or get a new Jetpack site from https://jurassic.ninja/ and register with Wordpress.com 
* Purchase jetpack Security
* Visit the following URL to add scan product directly to the cart. The plans page has checks that don't allow you to add.  https://**[TEST_DOMAIN]**/checkout/**[TEST_SITE]**/jetpack_scan?site=**[TEST_SITE]** 
* You see the following warning at the top of the shopping cart.
![](https://cloudup.com/cxiFh0uKrzO+)

Repeat same tests for Backup and Antispam features. 

These tests are to show a warning message if the user is trying to purchase a feature to which they already have access to  because of already purchased plan.


Related to p4TIVU-a66-p2
